scipy/interpolate/_ndgriddata.py

```diff
@@ -116,8 +122,16 @@ def __call__(self, *args):
         xi = _ndim_coords_from_arrays(args, ndim=self.points.shape[1])
         xi = self._check_call_shape(xi)
         xi = self._scale_x(xi)
-        dist, i = self.tree.query(xi)
-        return self.values[i]
+        dist, i = self.tree.query(xi, distance_upper_bound=self.query_max_dist)
+        if np.isfinite(self.query_max_dist):
+            # dist will be infinite for points where NN returned no neighbors. set those to nan, and mask in arr.
```

Comment:
Is this what `kdtree` returns? Would be great to stress it in the comment if it's indeed the case.

Reply:
Added some more content in the text to stress this point and clarify.